docs/tutorial.md

```diff
@@ -0,0 +1,721 @@
+
+7. Follow the prompts appropriate to your situation.
+
+    If you are restoring an existing ID, you may see a prompt about your user
+    being nameless, ignore it. At this point you have only a single application
+    on your test server.  So, you should see this single application, with your
+    own `blockstack.id` display name, once you are signed in:
+
+    ![](images/running-app.png)
+
+
+### Add a redirect end point to your application
+
+When a user opens the webapp from the Blockstack browser on an iOS phone,
+you want the web app to redirect the user to your iOS application. The work
+you do here will allow it.
+
+1. From the terminal command line, change directory to the root of your web
+application directory.
+
+2. Use the `touch` command to add a redirect endpoint to your application.
+
+   This endpoint on the web version of your app will redirect iOS users back
+   to your mobile app.
+
+    ```bash
+    $ touch public/redirect.html
+    ```
+
+3. Open `redirect.html` and add code to the endpoint.
+
+    ```
+    <!DOCTYPE html>
+    <html>
+    <head>
+      <title>Hello, Blockstack!</title>
+      <script>
+      function getParameterByName(name) {
+        var match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
+        return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
+      }
+
+      var authResponse = getParameterByName('authResponse')
+      window.location="myblockstackapp:" + authResponse
+      </script>
+    <body>
+    </body>
+    </html>
+    ```
+
+    Blockstack apps are identified by their domain names.  The endpoint will
+    receive a get request with the query parameter `authResponse=XXXX` and
+    should redirect the browser to `myblockstackapp:XXXX`.
+
+    `myblockstackapp:` is custom protocol handler. The handler should be unique
+    to your application. Your app's web-based authentication uses this handler
+    to redirect the user back to your iOS app. Later, you'll add a reference
+    to this handler in your iOS application.
+
+5. Close and save the `redirect.html` file.
+6. Ensure your Blockstack compiles successfully.
+
+## Build the hello-blockstack-ios
+
+Now, you build an iOS application that can access and run your Blockstack web
+application on a mobile device.
+
+### Create an XCode Project
+
+This tutorial uses XCode 9.3, you can use another version but be aware that some
+menu items and therefore these procedures may be differœent on your version.
+
+1. Launch the XCode interface.
+2. Choose **Create new XCode project**.
+3. Select **iOS**.
+4. Select **Single View Page**.
+
+   ![](images/single-view-app.png)
+
+5. **Choose options for your new project** for your project.
+
+	 ![](images/choose-new-options.png)
+
+6. Press **Next**.
+
+   The system prompts you for a location to store your code.
+
+7. Save your project.
+8. Close XCode.
+
+
+### Add and edit a Podfile
+
+To use CocoaPods you need to define the XCode target to link them to.
```

Comment:
Would include something about what cocoa pods is: “Cocoapods is a dependency manager for Swift and Objective-C Cocoa projects. It’s a simple, user friendly way to use libraries from the community in your project.”